src/_pytest/main.py

```diff
@@ -175,6 +178,16 @@ def pytest_addoption(parser):
     )
 
 
+def validate_basetemp(path):
+    # GH 7119
+    cwd = pathlib.Path.cwd()
+
+    if path == "" or path == "." or str(cwd).startswith(path):
```

Comment:
Checking for string-prefix (`startswith`) isn't correct with paths. [`Path.parents`](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.parents) would be a better choice. You'd probably also want to make `path` absolute and resolved for comparison purposes.

Reply:
Agreed, and probably do check against absolute path *and* a resolved path to be sure, something like (untested):

```python
def validate_basetemp(path):

    msg = "basetemp must not be empty, the current working directory or any parent directory of it"
    
    # empty path
    if not path:
        raise argparse.ArgumentTypeError(msg)

    def is_ancestor(base, path):
        if base == path:
            return True
        for p in base.parents:
            if p == path:
                return True
        return False
    
    # check if path is an ancestore
    if is_ancestor(Path.cwd(), Path(path).absolute()):
        raise argparse.ArgumentTypeError(msg)

    # check symlinks for ancestors
    if is_ancestor(Path.cwd().resolve(), Path(path).resolve()):
        raise argparse.ArgumentTypeError(msg)
        
    return path

    
```

Reply:
> Checking for string-prefix (startswith) isn't correct with paths.

Yes. I agree. Thanks for feedback @bluetech and @nicoddemus. I updated `validate_basetemp` method.